dll/win32/iernonce/registry.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * PROJECT:     ReactOS system libraries
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Functions to read RunOnceEx registry.
+ * COPYRIGHT:   Copyright 2021 He Yang <1160386205@qq.com>
+ */
+
+
+#include <cassert>
+#include "registry.h"
+#include <winreg.h>
+
+
+typedef BOOL(*REGENUMCALLBACK)(CRegKeyEx &hKey, LPCWSTR lpObjectName, DWORD cchszName, PVOID Param);
+
+enum class REG_ENUM_TYPE
+{
+    SUBKEY,
+    VALUE
+};
+
+
+LONG CRegKeyEx::EnumValueName(
+    DWORD iIndex,
+    LPTSTR pszName,
+    LPDWORD pnNameLength)
```

Comment:
```suggestion
LONG CRegKeyEx::EnumValueName(
    _In_ DWORD iIndex,
    _Out_ LPTSTR pszName,
    _Out_ LPDWORD pnNameLength)
```

Reply:
Looks great! I think probably `_Inout_` is better for `pnNameLength`? XD
By the way, I wonder when should I write SAL annotations? Should I write annotations whenever possible?